### PR TITLE
Ignore join build side exchange type in tests

### PR DESCRIPTION
**Pull request by lukasz-stec**
This change aims at reducing test over-specification for tests that assert join plans with specific build side exchange type.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExchangeMatcher.java [hunk 1]
```diff
@@ -37,15 +37,15 @@
         implements Matcher
 {
     private final ExchangeNode.Scope scope;
-    private final ExchangeNode.Type type;
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExchangeMatcher.java:40]: just make it `Optional<ExchangeNode.Type>`. Would that work?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java [hunk 2]
```diff
@@ -165,7 +165,7 @@ public void testPrunePartitionLikeFilter()
                                                 project(
                                                         filter("\"like\"(L_STR_PART, \"$like_pattern\"('t%'))",
                                                                 tableScan("table_str_partitioned", Map.of("L_INT_COL", "int_col", "L_STR_PART", "str_part"))))),
-                                        exchange(LOCAL, REPARTITION,
```
Comment (sopel39) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java:168]: you could also use `exchange(io.trino.sql.planner.assertions.PlanMatchPattern...)`. It's not really the point of this test to match exchange type.

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExchangeMatcher.java (hunk 1 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExchangeMatcher.java:40]: changed to Optional

**Review by lukasz-stec — COMMENTED** [review]
commentes addressed

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExchangeMatcher.java [hunk 3]
```diff
@@ -37,15 +37,15 @@
...
+    private final Optional<ExchangeNode.Type> type;
     private final List<Ordering> orderBy;
     private final Set<String> partitionedBy;
     private final Optional<List<List<String>>> inputs;
 
-    public ExchangeMatcher(ExchangeNode.Scope scope, ExchangeNode.Type type, List<Ordering> orderBy, Set<String> partitionedBy, Optional<List<List<String>>> inputs)
+    public ExchangeMatcher(ExchangeNode.Scope scope, Optional<ExchangeNode.Type> type, List<Ordering> orderBy, Set<String> partitionedBy, Optional<List<List<String>>> inputs)
     {
         this.scope = scope;
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExchangeMatcher.java:47]: nit: you could also add `rnn` for scope

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java [hunk 4]
```diff
@@ -632,28 +632,59 @@ public static PlanMatchPattern exchange(PlanMatchPattern... sources)
         return node(ExchangeNode.class, sources);
     }
 
+    public static PlanMatchPattern exchange(ExchangeNode.Scope scope, PlanMatchPattern... sources)
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:635]: just add one method that you require in test.
nit: Ideally, we have builder like: `io.trino.sql.planner.assertions.RowNumberMatcher.Builder`

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java [hunk 5]
```diff
@@ -1077,10 +1077,10 @@ public void testJoinPushdown()
 
         String notDistinctOperator = "IS NOT DISTINCT FROM";
         List<String> nonEqualities = Stream.concat(
-                Stream.of(JoinCondition.Operator.values())
-                        .filter(operator -> operator != JoinCondition.Operator.EQUAL)
-                        .map(JoinCondition.Operator::getValue),
-                Stream.of(notDistinctOperator))
+                        Stream.of(JoinCondition.Operator.values())
```
Comment (sopel39) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java:1080]: that seems like formatting fix, extract to separate commit

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java [hunk 6]
```diff
@@ -165,7 +164,7 @@ public void testPrunePartitionLikeFilter()
...
```
Comment (sopel39) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java:168]: now that you kept new matcher, it makes sense to use 
```
exchange(LOCAL,
  ...
```
e.g it's either you keep the matcher and use it here, or remove matcher and just match exchange

**Review by sopel39 — COMMENTED** [review]
lgtm % comments

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java (hunk 4 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:635]: I removed not unnecessary methods.

> nit: Ideally, we have builder like: io.trino.sql.planner.assertions.RowNumberMatcher.Builder

IMO at this point, it's better to have static methods and introduce builder when there is more variability in the matcher usage.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java (hunk 6 above)
Comment (lukasz-stec) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java:168]: ok, lets match LOCAL here

**Review by lukasz-stec — COMMENTED** [review]
comments addressed + rebased on the master

**Review by sopel39 — APPROVED** [review]
